programs/zstdcli.c

```diff
@@ -615,8 +615,21 @@ static int init_cLevel(void) {
     return ZSTDCLI_CLEVEL_DEFAULT;
 }
 
-#define ZSTD_NB_STRATEGIES 9
+#define NEXT_FIELD(ptr) {      \
```

Comment:
Would it make sense to handle `--flag ARG` and `--flag=ARG` with the same macro? That way we don't need to list each argument twice.

We could use `longCommandWArg()` to increment `argument` to the `{'=', '\0'}` character, then take care of that case within the macro.

```c
#define NEXT_FIELD(ptr) do { \
    if (argument[0] == '=') { ++argument; ptr = *argument; break; } \
    /* same as before */ \
} while (0);

...

if (longCommandWArg(&argument, "--flag")) { char char* arg; NEXT_FIELD(ptr); continue; }
```

Reply:
Yes, that's a good follow-up,
since it would simplify managing both `--long-command`.

Depending on how it's implemented, it might also introduce _new_ syntax, which is so far invalid
such as`-o=FILE`, which is so far not allowed because `-o` is a short command.

Not sure what to think about it.

Reply:
That should be manageable. If `argument` isn't incremented with `-o` then `argument[0] == 'o'`. Only commands that increment `argument`, like with `longCommandWArg()`.